File: src/main/java/com/tombnb22/service/ListingAmenityService.java
package com.tombnb22.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.tombnb22.domain.ListingAmenity;
import com.tombnb22.dto.ListingAmenityDTO;
import com.tombnb22.dto.ListingAmenitySearchDTO;
import com.tombnb22.dto.ListingAmenityPageDTO;
import com.tombnb22.dto.ListingAmenityConvertCriteriaDTO;
import com.tombnb22.service.GenericService;
import com.tombnb22.dto.common.RequestDTO;
import com.tombnb22.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ListingAmenityService extends GenericService<ListingAmenity, Integer> {

	List<ListingAmenity> findAll();

	ResultDTO addListingAmenity(ListingAmenityDTO listingAmenityDTO, RequestDTO requestDTO);

	ResultDTO updateListingAmenity(ListingAmenityDTO listingAmenityDTO, RequestDTO requestDTO);

    Page<ListingAmenity> getAllListingAmenitys(Pageable pageable);

    Page<ListingAmenity> getAllListingAmenitys(Specification<ListingAmenity> spec, Pageable pageable);

	ResponseEntity<ListingAmenityPageDTO> getListingAmenitys(ListingAmenitySearchDTO listingAmenitySearchDTO);
	
	List<ListingAmenityDTO> convertListingAmenitysToListingAmenityDTOs(List<ListingAmenity> listingAmenitys, ListingAmenityConvertCriteriaDTO convertCriteria);

	ListingAmenityDTO getListingAmenityDTOById(Integer listingAmenityId);







}

File: tombnb22_ui/src/services/ListingAmenityService.js
import http from "../http-common";

class ListingAmenityService {
  getAllListingAmenitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/listingAmenity/listingAmenitys`, searchDTO);
  }

  get(listingAmenityId) {
    return this.getRequest(`/listingAmenity/${listingAmenityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/listingAmenity?field=${matchData}`, null);
  }

  addListingAmenity(data) {
    return http.post("/listingAmenity/addListingAmenity", data);
  }

  update(data) {
  	return http.post("/listingAmenity/updateListingAmenity", data);
  }
  
  uploadImage(data,listingAmenityId) {
  	return http.postForm("/listingAmenity/uploadImage/"+listingAmenityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ListingAmenityService();
